### DISCUSSION TITLE
Second migration alter my primary key

### DISCUSSION THREAD

**Discussion by movaldivia** [sqlalchemy/alembic#953] [Usage Questions] [answered]
**Describe the bug**
I started a new db. I migrated it and then did another migration where the autogenerate alter my primary key.

First migration:

![image](https://user-images.githubusercontent.com/24725124/137398273-e0fafaf4-b373-4ced-9c57-8b5cc838f529.png)

Second migration without updating the model:

![image](https://user-images.githubusercontent.com/24725124/137398349-4a24cce3-025b-4d67-a33f-37fbb74fcd1e.png)

So I get this error:

[SQL: ALTER TABLE ********* ALTER COLUMN id DROP NOT NULL]

Using:
postgres
wls

**Comment by zzzeek** [comment]
hi there -

that's not an error message, that looks like the bottom part of one.       nothing here is illustrated going wrong and there's no detail we can work with.

if you haven't used Alembic before perhaps this can be a discussion where you can share more details of your configuration and such.

**Comment by CaselIT** [comment, marked as answer]
Hi,

The issue seems to be that you are setting nullable=True on a primary key in the sqlalchemy table.
Postgresql does not support that: https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-PRIMARY-KEYS

Alembic is seeing a difference there and is trying to set the column nullable.

To solve this you should remove nullable=True from the table definition

**Reply by movaldivia** [reply]
You are right! Ty so much!
